python/example/network_ring.py

```diff
@@ -109,9 +109,7 @@ def global_properties(self, kind):
 recipe = ring_recipe(ncells)
 
 # (12) Create an execution context using all locally available threads, domain decomposition and simulation
-context = arbor.context("avail_threads")
-decomp = arbor.partition_load_balance(recipe, context)
-sim = arbor.simulation(recipe, decomp, context)
+sim = arbor.simulation(recipe, context=arbor.context("avail_threads"))
```

Comment:
```suggestion
context = arbor.context("avail_threads")
sim = arbor.simulation(recipe, context=context)
```

Reply:
Cleaner, more uniform.

Reply:
It's an extra line. Unacceptable.

Reply:
What about making the load balancer the third arg, and optional? It's the least likely element that users will customize. We can then forego with the still somewhat verbose `context=arbor.context`, but say instead `sim = arbor.simulation(recipe, arbor.context("avail_threads"))`

Reply:
You can currently set `context`, `domain_decomposition` or both. So, I opt for the parallel to our C++ API
and keep the order.

Reply:
Can we swap it there too? The argument applies there too: least likely last.

Reply:
That requires more thinking, breaking an API should be done carefully (even in 0.x.y). Note that this _here_ is 
not a breaking change, the old way still works. Consequently, I'd like to separate it into its own PR.
Finally, I am less than sure that the decomposition is the least likely.

Reply:
I can't see how you'd be able to set a nonstandard domdec without supplying the context, so in that sense domdec is the least likely to be modified (any customization also requires a context, but not vice versa).

And you already add a new constructor to `arbor/include/simulation.hpp` with this different ordering, when is then used to update the cpp examples. You had to change the ordering there for nearly similar reasons: you can create a load balancer without having a context first.

So, I remain of the opinion to switch to the shape of that new constructor (because it is the most intuitive shape).

Reply:
Again, leave that discussion for a separate PR/Issue. Breaking changes should be done is isolation and this is big already. Good catch on the C++ side though, the order there should be reversed.